### PR TITLE
Devengage 1142 skill groups v4

### PR DESCRIPTION
**Pull request by carnellj**

### CODE REVIEW

File: GNUmakefile [hunk 1]
```diff
@@ -7,7 +7,7 @@ BIN_NAME=terraform-provider-genesyscloud
 BIN_PATH=${DIST_DIR}/${BIN_NAME}
 
 PLUGINS_DIR=~/.terraform.d/plugins
-PLUGIN_PATH=genesys.com/mypurecloud/genesyscloud
+PLUGIN_PATH=genesys.local.com/mypurecloud/genesyscloud
```
Comment (ronanwatkins) [inline GNUmakefile:10]: Guessing you didn't mean to include this

File: docs/data-sources/routing_skill_group.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `name` (String) Skill group name.
+
+### Optional
+
+- `description` (String) Description of the skill group.
```
Comment (ronanwatkins) [inline docs/data-sources/routing_skill_group.md:30]: Was this file auto generated? I don't think these should be here

File: genesyscloud/data_source_genesyscloud_routing_skill_group_test.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+		skillGroupDataSource = "routing-skill-group-data"
+		skillGroupName       = "Skillgroup" + uuid.NewString()
+	)
+
+	config := generateRoutingSkillGroupResource(
+		skillGroupResource,
+		skillGroupName,
+	) + generateRoutingSkillGroupDataSource(skillGroupDataSource, "genesyscloud_routing_skill_group."+skillGroupResource+".name", "genesyscloud_routing_skill_group."+skillGroupResource)
+
+	log.Printf(config)
```
Comment (ronanwatkins) [inline genesyscloud/data_source_genesyscloud_routing_skill_group_test.go:24]: Can remove this

File: genesyscloud/resource_genesyscloud_routing_skill_group.go [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+	Entities []struct {
+		ID   string `json:"id"`
+		Name string `json:"name"`
+	}
+	NextURI     string `json:"nextUri"`
+	SelfURI     string `json:"selfUri"`
+	PreviousURI string `json:"previousUri"`
+}
+
+//TODO
```
Comment (ronanwatkins) [inline genesyscloud/resource_genesyscloud_routing_skill_group.go:40]: These comments like TODO and //Done can be removed

File: docs/resources/routing_skill_group.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+## API Usage
+The following Genesys Cloud APIs are used by this resource. Ensure your OAuth Client has been granted the necessary scopes and permissions to perform these operations:
+
+* [GET /api/v2/routing/skillgroups](https://developer.genesys.cloud/platform/preview-apis#get-api-v2-routing-skillgroups)
+* [POST /api/v2/routing/skillgroups](https://developer.genesys.cloud/platform/preview-apis#post-api-v2-routing-skillgroups)
+* [PATCH /api/v2/routing/skillgroups/{skillGroupId}](https://developer.genesys.cloud/platform/preview-apis#patch-api-v2-routing-skillgroups--skillGroupId-)
+* [DELETE /api/v2/routing/skillgroups/{skillGroupId}](https://developer.genesys.cloud/platform/preview-apis#delete-api-v2-routing-skillgroups--skillGroupId-)
+
+**NOTE: The Routing Skill Groups feature is currently in beta and preview.  This means while the APIs are available in preview (they can change).  The feature is in a closed beta.  You must be part of the beta in order to use this resource.  If you are not part of the beta, ths resource will not function properly for you.  Please contact Chris Bohlin (chris.bohlin@genesys.com) to be added to the beta.
```
Comment (ronanwatkins) [inline docs/resources/routing_skill_group.md:19]: Looks like you've manually wrote stuff here. Usually we auto generate these files with `go generate` using the resource description, `resource.tf` and `apis.md`. If you put the right stuff into those files it should be able to pick them up and write this file.

File: genesyscloud/resource_genesyscloud_routing_skill_group_test.go [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+				),
+			},
+			{
+				// Import/Read
+				ResourceName:      "genesyscloud_routing_skill_group." + skillGroupResource,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+		CheckDestroy: testVerifySkillsDestroyed,
```
Comment (ronanwatkins) [inline genesyscloud/resource_genesyscloud_routing_skill_group_test.go:42]: Guessing you mean to call `testVerifySkillGroupDestroyed` here

File: GNUmakefile (hunk 1 above)
Comment (carnellj) [reply GNUmakefile:10]: Fixed

File: docs/resources/routing_skill_group.md (hunk 5 above)
Comment (carnellj) [reply docs/resources/routing_skill_group.md:19]: Oopps.  I never realized we auto-generated these files.  Any reason why we check the generated doc files into source control.  I will fix

File: genesyscloud/data_source_genesyscloud_routing_skill_group_test.go (hunk 3 above)
Comment (carnellj) [reply genesyscloud/data_source_genesyscloud_routing_skill_group_test.go:24]: Done

File: genesyscloud/resource_genesyscloud_routing_skill_group.go (hunk 4 above)
Comment (carnellj) [reply genesyscloud/resource_genesyscloud_routing_skill_group.go:40]: Done

File: genesyscloud/resource_genesyscloud_routing_skill_group_test.go (hunk 6 above)
Comment (carnellj) [reply genesyscloud/resource_genesyscloud_routing_skill_group_test.go:42]: Done

File: genesyscloud/resource_genesyscloud_routing_skill_group_test.go [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+func testVerifySkillGroupDestroyed(state *terraform.State) error {
+
+	// Get default config to set config options
+	config := platformclientv2.GetDefaultConfiguration()
+	routingAPI := platformclientv2.NewRoutingApi()
+	apiClient := &routingAPI.Configuration.APIClient
+
+	// TODO Once this code has been released into the public API we should fix this and use the SDK
+	err := config.AuthorizeClientCredentials(os.Getenv("GENESYS_CLOUD_CLIENT_ID"), os.Getenv("GENESYS_CLOUD_CLIENT_SECRET"))
```
[the remark this answers is not in the document]
Comment (carnellj) [reply genesyscloud/resource_genesyscloud_routing_skill_group_test.go:63]: I am bit confused. Are you saying I dont need to authorize the client.  The testVerifyProcessAutomationTrigger is using the SDK so the credentials do not have to be explicitly authorized.  Can you provide some more context here?

File: genesyscloud/resource_genesyscloud_routing_skill_group_test.go (hunk 6 above)
Comment (carnellj) [reply genesyscloud/resource_genesyscloud_routing_skill_group_test.go:42]: Code by cut and paste has its risks

File: docs/resources/routing_skill_group.md (hunk 5 above)
Comment (carnellj) [reply docs/resources/routing_skill_group.md]: Fixed

**Comment by ronanwatkins** [comment]
> You'll need to run `go generate` to create some example files and check them in

Docs aren't in sync

File: genesyscloud/resource_genesyscloud_routing_skill_group_test.go [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+			{
+				// Create
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("genesyscloud_routing_skill_group."+skillGroupResource, "name", skillGroupName),
+					resource.TestCheckResourceAttr("genesyscloud_routing_skill_group."+skillGroupResource, "description", skillGroupDescription),
+					testDefaultHomeDivision("genesyscloud_routing_skill_group."+skillGroupResource),
+				),
+			},
+			{
```
Comment (ronanwatkins) [inline genesyscloud/resource_genesyscloud_routing_skill_group_test.go:71]: Will need to update the resource attributes to make sure that all works

File: genesyscloud/resource_genesyscloud_routing_skill_group_test.go [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+
+func TestAccResourceRoutingSkillGroupBasic(t *testing.T) {
+	var (
+		skillGroupResource     = "testskillgroup1"
+		skillGroupName1        = "SkillGroup1" + uuid.NewString()
+		skillGroupDescription1 = "Description1" + uuid.NewString()
+		skillGroupName2        = "SkillGroup2" + uuid.NewString()
+		skillGroupDescription2 = "Description2" + uuid.NewString()
+		divHomeRes             = "auth-division-home"
+		divHomeName            = "Home"
```
[the remark this answers is not in the document]
Comment (carnellj) [reply genesyscloud/resource_genesyscloud_routing_skill_group_test.go]: fixed

**Pull request merged**